drivers/SmartThings/matter-switch/src/test/test_multi_switch.lua

```diff
@@ -152,4 +211,26 @@ test.register_message_test(
   }
 )
 
+test.register_message_test(
+  "On command to component switch should send the appropriate commands for devices with non-sequential endpoints",
+  {
+    {
+      channel = "capability",
+      direction = "receive",
+      message = {
+        mock_3switch_non_sequential.id,
+        { capability = "switch", component = "switch3", command = "on", args = { } }
+      }
+    },
+    {
+      channel = "matter",
+      direction = "send",
+      message = {
+        mock_3switch_non_sequential.id,
+        clusters.OnOff.server.commands.On(mock_3switch_non_sequential, 15) -- switch 3 is on endpoint 15
```

Comment:
This is correct, but it got a little thinking to realize that we sequentially increase the switch numbers, internally, as opposed to using the endpoint_id number.

Reply:
In our profiles, the switch components are numbered sequentially (except the first switch component is called "main", otherwise it continues on as "switch2", "switch3", etc.). 

The purpose of these changes is that the endpoint number does not need to match the component number. It can be a little confusing, but the idea is that when mapping a component to an endpoint, the component will map to the next lowest endpoint that hasn't been mapped yet. So for example, in a case where we have endpoints 10, 12, and 15, the mapping would be:
"main" -> 10
"switch2" -> 12
"switch3" -> 15

I will add some additional documentation to make this clear. Thanks for pointing it out!

Reply:
Thank you @ctowns !

Reply:
Added a comment in the `create_endpoint_to_component_map()` function to help explain the reasoning behind the component numbering in a9e0c3ac0e37798c0fd3f4927ce6b4ad80d88fc0